llm/predictor.py

```diff
@@ -703,6 +723,526 @@ def _infer(self, inputs: dict[str, paddle.Tensor]):
+                dtype=self.dtype,
+            )
+            pre_cache_mask[:, :, :, : self.pre_cache_length] = 1
+            pre_cache_mask[:, :, :, self.pre_cache_length :] = paddle.tril(
+                paddle.ones(shape=[config.batch_size, 1, config.src_length, config.src_length], dtype=self.dtype)
+            )
+            self.inputs["src_mask"] = (pre_cache_mask - 1) * 1e4
+
+        if config.use_cachekv_int8 == "dynamic":
+            self.k_quant_scales = [
+                paddle.zeros([config.batch_size, self.num_attention_heads], dtype="float32")
+                for _ in range(self.num_layers)
+            ]
+            self.v_quant_scales = [
+                paddle.zeros([config.batch_size, self.num_attention_heads], dtype="float32")
+                for _ in range(self.num_layers)
+            ]
+            self.k_dequant_scales = [
+                paddle.zeros([config.batch_size, self.num_attention_heads], dtype="float32")
+                for _ in range(self.num_layers)
+            ]
+            self.v_dequant_scales = [
+                paddle.zeros([config.batch_size, self.num_attention_heads], dtype="float32")
+                for _ in range(self.num_layers)
+            ]
+
+        # not update
+        self.inputs["cache_kvs"] = self.cache_kvs
+
+        self.inputs["pre_ids"] = paddle.full([config.batch_size, self.total_max_length], -1, dtype="int64")
+        self.inputs["bad_tokens"] = paddle.to_tensor(
+            [
+                -1,
+            ],
+            dtype="int64",
+        )
+        self.inputs["penalty_score"] = paddle.full(shape=[config.batch_size, 1], fill_value=1.0, dtype="float32")
+        self.inputs["frequency_score"] = paddle.full(shape=[config.batch_size, 1], fill_value=0.0, dtype="float32")
+        self.inputs["presence_score"] = paddle.full(shape=[config.batch_size, 1], fill_value=0.0, dtype="float32")
+
+        if config.use_cachekv_int8 == "dynamic":
+            self.inputs["k_quant_scales"] = self.k_quant_scales
+            self.inputs["v_quant_scales"] = self.v_quant_scales
+            self.inputs["k_dequant_scales"] = self.k_dequant_scales
+            self.inputs["v_dequant_scales"] = self.v_dequant_scales
+
+        if config.benchmark:
+            min_length = config.max_length
+        else:
+            min_length = 2
+        self.inputs["min_length"] = paddle.full(shape=[config.batch_size, 1], fill_value=min_length, dtype="int64")
+        self.inputs["max_length"] = paddle.full(
+            shape=[config.batch_size, 1], fill_value=config.max_length, dtype="int64"
+        )
+        self.inputs["stop_nums"] = paddle.full(shape=[1], fill_value=config.batch_size, dtype="int64")
+        tmp_position_ids = paddle.arange(self.total_max_length).reshape((1, -1))
+        self.inputs["rope_emb"] = self._get_rotary_position_embedding(tmp_position_ids, self.head_dim)
```

Comment:
```suggestion
        self.inputs["rope_emb"] = self._get_rotary_position_embedding(
            paddle.arange(self.total_max_length).reshape((1, -1)), 
            self.head_dim
        )
```

下同，尽量避免 tmp_* 这种临时变量。